### PR TITLE
OCPBUGS-33962: [release-4.16] Remove vSphere Terraform

### PR DESCRIPTION
**Pull request by jcpowermac**

### CODE REVIEW

**Comment by jcpowermac** [comment]
Looked at CI past cap{i,v} stage

/assign @patrickdillon @r4f4

**Comment by patrickdillon** [comment]
/hold 
TRT is reverting capi installs in #8434 until we can fix timeouts on aws bootstrap ssh rule cleanup

**Comment by jcpowermac** [comment]
https://github.com/openshift/installer/pull/8451
merged so removed the hold

File: pkg/infrastructure/platform/platform.go [hunk 1]
```diff
@@ -95,7 +94,7 @@ func ProviderForPlatform(platform string, fg featuregates.FeatureGate) (infrastr
 		if types.ClusterAPIFeatureGateEnabled(platform, fg) {
 			return clusterapi.InitializeProvider(vspherecapi.Provider{}), nil
 		}
-		return terraform.InitializeProvider(vsphere.PlatformStages), nil
+		return nil, fmt.Errorf("installing on vSphere via terraform is no longer supported")
```
Comment (r4f4) [inline pkg/infrastructure/platform/platform.go:97]: Now that capi/vsphere is the default, we could just drop this error and remove the `types.ClusterAPIFeatureGateEnabled` check. We can do that as a follow-up.

**Review by r4f4 — COMMENTED** [review]
/approve

**Comment by jcpowermac** [comment]
vsphere failure caused by too many 'etcd took long messages'

**Pull request merged**